packages/perseus/src/widgets/interactive-graphs/mafs-graph.tsx

```diff
@@ -73,15 +78,25 @@ export const MafsGraph = React.forwardRef<
         props,
         initializeGraphState,
     );
-    const prevState = useRef<InteractiveGraphState>(state);
 
+    const prevState = useRef<InteractiveGraphState>(state);
     useEffect(() => {
         if (prevState.current !== state) {
             props.onChange({graph: state});
         }
         prevState.current = state;
     }, [props, state]);
 
+    const {snapStep, range} = props;
+
+    useEffect(() => {
+        dispatch(changeSnapStep(snapStep));
+    }, [snapStep]);
+
+    useEffect(() => {
+        dispatch(changeRange(range));
+    }, [range]);
```

Comment:
I wonder if, to prevent unnecessary re-renders, the individual numbers in `snapStep` and `range` could be listed as deps here:

```ts
    useEffect(() => {
        dispatch(changeSnapStep(snapStep));
    }, snapStep);

    useEffect(() => {
        dispatch(changeRange(range));
    }, range.flat());
```

We could also maybe `useLayoutEffect` so the state update happens before React renders anything to the DOM.

Reply:
I did what you asked but in a more verbose way. The linter didn't seem happy with your suggestion and I think being a little more verbose makes the intent a little more clear. It's a good suggestion though, solid improvement.